test/fork/migration-v2.js

```diff
@@ -61,6 +80,33 @@ const submitGovernanceProposal = async (categoryId, actionData, signers, gv) =>
   assert.equal(proposal[2].toNumber(), 3);
 };
 
+// This works with the V2 version of Governance.sol
+const submitGovernanceProposalV2 = async (categoryId, actionData, signers, gv) => {
+  const id = await gv.getProposalLength();
+
+  await gv.connect(signers[0]).createProposal('', '', '', 0);
+  await gv.connect(signers[0]).categorizeProposal(id, categoryId, 0);
+  await gv.connect(signers[0]).submitProposalWithSolution(id, '', actionData);
+
+  for (let i = 0; i < signers.length; i++) {
+    await gv.connect(signers[i]).submitVote(id, 1, []);
```

Comment:
@danoctavian to avoid having 2 separate paths here for V1 and V2, I suggest making the following adjustment in Governance.sol:
1. rename the current `submitVote` function to `submitVoteWithStakingPoolIds` - or similar
2. create a new `submitVote` with the same signature as the one in V1 `function submit Vote (uint _proposalId, uint _solutionChosen) external` that calls into `submitVoteWithStakingPoolIds` with `[]` as the `managedStakingPoolIds` param